src/test/kotlin/org/rust/ide/inspections/lints/RsUnreachableCodeInspectionTest.kt

```diff
@@ -0,0 +1,218 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.lints
+
+import org.rust.MockAdditionalCfgOptions
+import org.rust.ProjectDescriptor
+import org.rust.WithStdlibRustProjectDescriptor
+import org.rust.ide.inspections.RsInspectionsTestBase
+
+class RsUnreachableCodeInspectionTest : RsInspectionsTestBase(RsUnreachableCodeInspection::class) {
+    fun `test straightline with return`() = checkByText("""
+        fn main() {
+            let x = 1;
+            return;<warning descr="Unreachable code">
+            let y = 2;
+            let z = 3;</warning>
```

Comment:
Can we exclude first PsiWhitespace during merging ranges?

Reply:
Yes, I think we can, but I intentionally leave it as is for now.
Do you think it would be better to exclude it? Maybe there are some real cases when it is necessary for good UI/UX?

Reply:
> some real cases when it is necessary for good UI/UX?

When using "next error" (F2) it will navigate to the correct line

Reply:
Fixed in a separate temporary commit for more convenience during review. Please let me know if the fix looks good to you, and I will finally rebase the commits

Reply:
Yes, fix looks good